doc/subsystems/networking/qemu_setup.rst

```diff
@@ -143,6 +143,8 @@ used, the following command should be run as root:
 
 Additionally, IPv4 forwarding should be enabled on host, and you may need to
 check that other firewall (iptables) rules don't interfere with masquerading.
+Remember also to set static gateway address via CONFIG_NET_APP_MY_IPV4_GW in
```

Comment:
I'd suggest adding xref markup for the config option (grep doctree for CONFIG_ to see the exact syntax).

Also, maybe move this to the start of the section (like, first list Zephyr requirement for access the Inet, then host's)?

Reply:
Ah, also don't forget that CONFIG_NET_APP_MY_IPV4_GW applies to net_app helper only. (So, to not say false things, probably should mention that too.)

Reply:
So, there're 2 levels of native networking API: a) original native networking API per se; b) net_app helper API on top of it. As the name suggests, CONFIG_NET_APP_MY_IPV4_GW applies to the latter. I don't know what the future plans are (e.g. almost all in-tree samples are converted to net_app), but the above is the current situation, and net_app isn't flexible enough to cover all usecases.

All in all, I'd suggest to rewrite that as:

"""
= paragraph break =

To access the Internet from a Zephyr application using IPv4, a gateway should be set. It can be set via DHCP, or configured manually. For the applications using net_app facility (:option:`CONFIG_NET_APP`), set :option:`CONFIG_NET_APP_MY_IPV4_GW` to the IP address of gateway. For apps not using net_app facility, :c:func:`net_if_ipv4_set_gw` should be called at runtime.
"""

Reply:
Even better would be:

     To access the Internet from a Zephyr application using IPv4,
     a gateway should be set via DHCP or configured manually. 
     For applications using the :ref:`net_app_api` facility (with the config option
     :option:`CONFIG_NET_APP` enabled), 
     set the :option:`CONFIG_NET_APP_MY_IPV4_GW` option to the IP address
     of the gateway. For apps not using the :ref:`net_app_api` facility, set up the
     gateway by calling the
     :c:func:`net_if_ipv4_set_gw` at runtime.

This will create nice links to the config options (as you suggested) as well as to the documentation for the net_app APIs.